hass-addon-sunsynk-dev/run.py

```diff
@@ -203,18 +239,36 @@ def setup_sensors() -> None:
         else:
             msg[fstr].append(name)  # type: ignore
 
-        SENSORS.append(getfilter(fstr, sensor=sen))
+        filt = getfilter(fstr, sensor=sen)
+        SENSORS.append(filt)
+        filters[sen.id] = filt
 
         if isinstance(sen, NumberRWSensor):
-            startup_sens.update(d.id for d in sen.dependencies())
+            for dep in sen.dependencies():
+                sens_dependencies[dep.id].append(sen)
 
-    # Add any sensor dependencies to STARTUP_SENSORS
-    try:
-        STARTUP_SENSORS.clear()
-        STARTUP_SENSORS.extend(ALL_SENSORS[n] for n in startup_sens)
-    except KeyError as err:
-        _LOGGER.fatal("Invalid sensor as dependency - %s", err)
+    for sen_id, deps in sens_dependencies.items():
+        try:
+            sen = ALL_SENSORS.get(sen_id)
+        except KeyError as err:
+            _LOGGER.fatal("Invalid sensor as dependency - %s", err)
 
+        if sen_id not in sens and sen != RATED_POWER:  # Rated power does not change
```

Comment:
This will be true when the user does not add this sensor, which has dependants, to the add-on options.